app/models/otp_requests_tracker.rb

```diff
@@ -1,10 +1,9 @@
 class OtpRequestsTracker < ApplicationRecord
-  def self.find_or_create_with_phone(phone)
-    tries ||= 1
-    phone ||= phone.strip
-    phone_fingerprint ||= Pii::Fingerprinter.fingerprint(phone)
+  def self.find_or_create_with_phone_and_confirmed(phone, phone_confirmed)
```

Comment:
A bit of a nit, but thoughts on changing this up to take additional args as options, e.g.

```ruby
def self.find_or_create_with_phone(phone, phone_confirmed:)
  # ...
end
```